pkg/azuredisk/azure_common_windows.go

```diff
@@ -19,100 +19,69 @@ limitations under the License.
 package azuredisk
 
 import (
-	"encoding/json"
 	"fmt"
 	"strconv"
-	"strings"
 
 	"k8s.io/klog"
 
 	"k8s.io/kubernetes/pkg/util/mount"
+	"sigs.k8s.io/azuredisk-csi-driver/pkg/mounter"
 )
 
-func scsiHostRescan(io ioHandler, exec mount.Exec) {
-	cmd := "Update-HostStorageCache"
-	output, err := exec.Run("powershell", "/c", cmd)
-	if err != nil {
-		klog.Errorf("Update-HostStorageCache failed in scsiHostRescan, error: %v, output: %q", err, string(output))
+func formatAndMount(source, target, fstype string, options []string, m *mount.SafeFormatAndMount) error {
+	proxy, ok := m.Interface.(*mounter.CSIProxyMounter)
+	if !ok {
+		return fmt.Errorf("could not cast to csi proxy class")
 	}
+	return proxy.FormatAndMount(source, target, fstype, options)
 }
 
-// search Windows disk number by LUN
-func findDiskByLun(lun int, iohandler ioHandler, exec mount.Exec) (string, error) {
-	cmd := `Get-Disk | select number, location | ConvertTo-Json`
-	output, err := exec.Run("powershell", "/c", cmd)
-	if err != nil {
-		klog.Errorf("Get-Disk failed in findDiskByLun, error: %v, output: %q", err, string(output))
-		return "", err
+// TODO: Return error from scsiHostRescan. Modify for Linux as well.
```

Comment:
so even `scsiHostRescan` return error, we should still continue find disk lun, right?

Reply:
Yep. Actually the TODO was unnecessary here.